`test_restore_db_replica/test.py::test_query_after_restore_db_replica` is flaky

Example failure (from [here](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=85551&sha=latest&name_0=PR&name_1=Integration+tests+%28amd_asan%2C+old+analyzer%2C+4%2F6%29)):

```
>           assert node_1.query_with_retry(
                f"SELECT table FROM system.tables WHERE database='{exclusive_database_name}' ORDER BY table",
                retry_count=10,
                sleep_time=1,
                check_callback=lambda tables: tables == exists_table_name,
            )
E           assert ''
E            +  where '' = <bound method ClickHouseInstance.query_with_retry of <helpers.cluster.ClickHouseInstance object at 0x7f2683624070>>("SELECT table FROM system.tables WHERE database='repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz' ORDER BY table", retry_count=10, sleep_time=1, check_callback=<function test_query_after_restore_db_replica.<locals>.<lambda> at 0x7f268188dc60>)
E            +    where <bound method ClickHouseInstance.query_with_retry of <helpers.cluster.ClickHouseInstance object at 0x7f2683624070>> = <helpers.cluster.ClickHouseInstance object at 0x7f2683624070>.query_with_retry

test_restore_db_replica/test.py:280: AssertionError
------------------------------ Captured log call -------------------------------
2025-08-14 12:05:38.474000 [ 698 ] DEBUG : Executing query 
                CREATE DATABASE repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz
                ENGINE=Replicated("/clickhouse/repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz", '{shard}', '{replica}')
             on node1 (cluster.py:4100, query)
2025-08-14 12:05:38.638000 [ 698 ] DEBUG : Executing query 
                CREATE DATABASE repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz
                ENGINE=Replicated("/clickhouse/repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz", '{shard}', '{replica}')
             on node2 (cluster.py:4100, query)
2025-08-14 12:05:38.803000 [ 698 ] DEBUG : Executing query 
            CREATE TABLE repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz.exists_table_zoywis (n UInt32)
            ENGINE = ReplicatedMergeTree
            ORDER BY n PARTITION BY n % 10
            SETTINGS distributed_ddl_task_timeout=5;
         on node1 (cluster.py:4100, query)
2025-08-14 12:05:39.068000 [ 698 ] DEBUG : Executing query 
            INSERT INTO repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz.exists_table_zoywis SELECT number FROM numbers(1000)
         on node1 (cluster.py:4100, query)
2025-08-14 12:05:39.333000 [ 698 ] DEBUG : get_instance_ip instance_name=zoo1 (cluster.py:2241, get_instance_ip)
2025-08-14 12:05:39.335000 [ 698 ] DEBUG : http://localhost:None "GET /v1.46/containers/roottestrestoredbreplica-gw2-zoo1-1/json HTTP/11" 200 None (connectionpool.py:546, _make_request)
2025-08-14 12:05:39.335000 [ 698 ] DEBUG : get_kazoo_client: zoo1, ip:172.16.1.4, port:2181, use_ssl:False (cluster.py:3727, get_kazoo_client)
2025-08-14 12:05:39.336000 [ 698 ] INFO : Connecting to 172.16.1.4(172.16.1.4):2181, use_ssl: False (connection.py:650, _connect)
2025-08-14 12:05:39.337000 [ 698 ] DEBUG : Sending request(xid=None): Connect(protocol_version=0, last_zxid_seen=0, time_out=30000, session_id=0, passwd=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', read_only=None) (connection.py:312, _submit)
2025-08-14 12:05:39.339000 [ 698 ] INFO : Zookeeper connection established, state: CONNECTED (client.py:532, _session_callback)
2025-08-14 12:05:39.339000 [ 698 ] DEBUG : Sending request(xid=1): GetChildren(path='/clickhouse/repl_db_test_query_after_restore_db_replica_alter_table_with_exists_table_with_restart__njwfjz', watcher=None) (connection.py:312, _submit)
```

[Test case history](https://play.clickhouse.com/play?user=play#U0VMRUNUIGhlYWRfcmVmLCBiYXNlX3JlcG8sIHB1bGxfcmVxdWVzdF9udW1iZXIgYXMgcHIsIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX2R1cmF0aW9uX21zLzEwMDAgQVMgcnVudGltZV9zZWMsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMsIHJlcG9ydF91cmwsIHRlc3RfY29udGV4dF9yYXcKRlJPTSBjaGVja3MKV0hFUkUgMQogICAgQU5EIHB1bGxfcmVxdWVzdF9udW1iZXIgPSAwCiAgICBBTkQgaGVhZF9yZWYgPSAnbWFzdGVyJwogICAgQU5EIHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicsICdCUk9LRU4nKQogICAgQU5EIHRvRGF0ZShjaGVja19zdGFydF90aW1lKSA+ICcyMDI1LTAxLTAxJwogICAgQU5EIHRlc3RfbmFtZSBsaWtlICcldGVzdF9yZXN0b3JlX2RiX3JlcGxpY2ElJwogICAgLS1BTkQgdGVzdF9jb250ZXh0X3JhdyBMSUtFICclVW5leHBlY3RlZCBudW1iZXIgb2YgY29sdW1ucyBpbiByZXN1bHQgc2FtcGxlIGJsb2NrJScKT1JERVIgQlkgY2hlY2tfc3RhcnRfdGltZSBERVNDClNFVFRJTkdTIHJlYWRfdGhyb3VnaF9kaXN0cmlidXRlZF9jYWNoZT0w)

--- a/src/Databases/DatabaseReplicated.cpp
+++ b/src/Databases/DatabaseReplicated.cpp
@@ -1811,6 +1811,10 @@ void DatabaseReplicated::restoreDatabaseMetadataInKeeper(ContextPtr)
         LOG_DEBUG(log, "It seems that the metadata was restored previously: {}.", e.what());
     }
 
+    /// Force the database to recover to update the restored metadata
+    auto current_zookeeper = getContext()->getZooKeeper();
+    current_zookeeper->set(replica_path + "/digest", DatabaseReplicatedDDLWorker::FORCE_AUTO_RECOVERY_DIGEST);
+
     reinitializeDDLWorker();
 }
 

--- a/src/Databases/DatabaseReplicatedWorker.cpp
+++ b/src/Databases/DatabaseReplicatedWorker.cpp
@@ -49,7 +49,6 @@ namespace FailPoints
     extern const char database_replicated_delay_entry_execution[];
 }
 
-static constexpr const char * FORCE_AUTO_RECOVERY_DIGEST = "42";
 
 DatabaseReplicatedDDLWorker::DatabaseReplicatedDDLWorker(DatabaseReplicated * db, ContextPtr context_)
     : DDLWorker(

--- a/src/Databases/DatabaseReplicatedWorker.h
+++ b/src/Databases/DatabaseReplicatedWorker.h
@@ -42,6 +42,8 @@ public:
 
     bool isUnsyncedAfterRecovery() const { return unsynced_after_recovery; }
 
+    static constexpr const char * FORCE_AUTO_RECOVERY_DIGEST = "42";
+
 private:
     bool initializeMainThread() override;
     void initializeReplication() override;

